realm/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -95,10 +96,13 @@ public DynamicRealmObject(RealmObject obj) {
 
     /**
      * Returns the {@code boolean} value for a given field.
+     * If the field is nullable use {@link #isNull(String)} to check for {@code null} instead of using
```

Comment:
Nullable isn't great, but "not required" has even less meaning IMO as it require people actually knowing about the @Required annotation and fully understanding what it does. At least nullability as a concept is hopefully familiar to most.